lib/exabgp/configuration/neighbor/__init__.py

```diff
@@ -239,9 +246,9 @@ def post (self):
 		if neighbor.range_size > 1 and not neighbor.passive:
 			return self.error.set('can only use ip ranges for the peer address with passive neighbors')
 
-		if neighbor.peer_address.top() in self._neighbors:
-			return self.error.set('duplicate peer definition %s' % neighbor.peer_address.top())
-		self._neighbors.append(neighbor.peer_address.top())
+		if neighbor.name() in self._neighbors:
+			return self.error.set('duplicate peer definition %s' % neighbor.name())
```

Comment:
here we should go through all the peers and look at the src ip / dst ip / family defined, and in case of duplicate just warn the user - it should be enough to help them in case of mis-configuration but still allow your use case.